aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -0,0 +1,388 @@
+#pragma once
+
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Tensor.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/sparse/Macros.h>
+#include <ATen/ExpandUtils.h>
+#include <ATen/SparseTensorUtils.h>
+#include <iostream>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/arange.h>
+#include <ATen/ops/empty.h>
+#include <ATen/ops/_sparse_coo_tensor_with_dims_and_tensors.h>
+#include <ATen/ops/from_blob.h>
+#include <ATen/ops/result_type.h>
+#endif
+
+#ifdef GPUCC
+#define NAME "sparse_binary_op_intersection_cuda"
+#else
+#define NAME "sparse_binary_op_intersection_cuda"
+#endif
+
+#define BINARY_OP_API static constexpr FUNCAPI INLINE
```

Comment:
Is this one actually used anywhere? We don't generally have much use for `constexpr`

Reply:
It will be used with a follow-up that will merge last 3 kernel calls into one. I.e. when the binary op becomes a lambda operating over scalar_t.

Reply:
With respect to the one definition rule `inline` and `constexpr` are treated the same way.